tests/enable-gcp-services/idempotent-prune/.expected/results.yaml

```diff
@@ -0,0 +1,49 @@
+apiVersion: kpt.dev/v1
+kind: FunctionResultList
+metadata:
+  name: fnresults
+exitCode: 0
+items:
+  - image: gcr.io/kpt-fn/enable-gcp-services:unstable
+    exitCode: 0
+    results:
+      - message: pruned service
+        severity: info
+        resourceRef:
+          apiVersion: serviceusage.cnrm.cloud.google.com/v1beta1
+          kind: Service
+          name: proj1-service-bigquery
+        file:
+          path: service_proj1-service-bigquery.yaml
+      - message: pruned service
+        severity: info
+        resourceRef:
+          apiVersion: serviceusage.cnrm.cloud.google.com/v1beta1
+          kind: Service
+          name: proj1-service-compute
+        file:
+          path: service_proj1-service-compute.yaml
+      - message: pruned service
```

Comment:
We show `pruned service` for `proj1-service-redis` in `service_proj1-service-redis.yaml` and later we show `generated service` for the same resource. These 2 operations combined together is a noop.
IMO this behavior may be a little confusing for users, since we actually only have one real operation here which is pruning `proj1-service-bigquery`, but we show users 5 operations. It will become harder to read when there are more resources. e.g. I have 10 resources unchanged and only one resource get pruned, there will be 21 events in the resultsList, it will be hard for users to figure out what really going on.
Maybe we can have something called `recreated service` to handle this case.
@bharathkkb WDYT?

Reply:
great idea, I have implemented this